File: python/basic_grammar/operation/assignment_operator/addition.py
def addition_assignment_operator():
    a = 0
    for i in range(0, 5):
        a += 2
    return a

# def back_addition_assignment_operator():
#     a = 0
#     for i in range(0, 5):
#         a++
#     return a

def front_addition_assignment_operator():
    a = 0
    for i in range(0, 5):
        ++a
    return a


def decimal_addition_assignment_operator():
    a = 0
    for i in range(0, 5):
        a += 1.5
    return a


def negative_addition_assignment_operator():
    a = 0
    for i in range(0, 5):
        a += -2
    return a


def string_addition_assignment_operator():
    s = ""
    for i in range(0, 5):
        s += "A"
    return s


def array_addition_assignment_operator():
    a = []
    for i in range(0, 5):
        a += [i]
    return a


print(addition_assignment_operator())
print(front_addition_assignment_operator())
print(decimal_addition_assignment_operator())
print(negative_addition_assignment_operator())
print(string_addition_assignment_operator())
print(array_addition_assignment_operator())

File: typescript/basic_grammar/operation/assignment_operator/addition.ts
export const additionAssignmentOperator = () => {
  let a = 0
  for (let i = 0; i <  5; i++) {
    a += 2
  }
  return a
}

export const backAdditionAssignmentOperator = () => {
  let a = 0
  for (let i = 0; i < 5; i++) {
    a++
  }
  return a
}

export const frontAdditionAssignmentOperator = () => {
  let a = 0
  for (let i = 0; i < 5; i++) {
    ++a
  }
  return a
}

export const decimalAdditionAssignmentOperator = () => {
  let a = 0
  for (let i = 0; i < 5; i++) {
    a += 1.5
  }
  return a
}

export const negativeAdditionAssignmentOperator = () => {
  let a = 0
  for (let i = 0; i < 5; i++) {
    a += -2
  }
  return a
}

export const stringAdditionAssignmentOperator = () => {
  let s = ""
  for (let i = 0; i < 5; i++) {
    s += "A"
  }
  return s
}

console.log(additionAssignmentOperator())
console.log(backAdditionAssignmentOperator())
console.log(frontAdditionAssignmentOperator())
console.log(decimalAdditionAssignmentOperator())
console.log(negativeAdditionAssignmentOperator())
console.log(stringAdditionAssignmentOperator())
